src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMissingLeadingAsteriskCheck.java

```diff
@@ -0,0 +1,210 @@
+ *  * Asterisks on first and last lines are optional.
+ *  * *&#47;
+ * class Asterisks {} // ok
+ *
+ * &#47;** No asterisks are required for single-line comments. *&#47;
+ * class SingleLine {} // ok
+ *
+ * &#47;** // violation on next blank line, javadoc has lines without leading asterisk.
+ *
+ *  *&#47;
+ * class BlankLine {}
+ *
+ * &#47;** Wrapped
+ *     single-line comment *&#47; // violation, javadoc has lines without leading asterisk.
+ * class Wrapped {}
+ *
+ * &#47;**
+ *  * &lt;pre&gt;
+ *     int value; // violation, javadoc has lines without leading asterisk.
+ *  * &lt;/pre&gt;
+ *  *&#47;
+ * class Code {}
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.missing.asterisk}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.38
+ */
+@StatelessCheck
+public class JavadocMissingLeadingAsteriskCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISSING_ASTERISK = "javadoc.missing.asterisk";
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.NEWLINE,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableJavadocTokens() {
+        return getRequiredJavadocTokens();
+    }
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return getRequiredJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode detailNode) {
+        DetailNode nextSibling = getNextNode(detailNode);
+
+        // Till https://github.com/checkstyle/checkstyle/issues/9005
+        // Due to bug in the Javadoc parser there may be phantom description nodes.
+        while (TokenUtil.isOfType(nextSibling.getType(),
+                JavadocTokenTypes.DESCRIPTION, JavadocTokenTypes.WS)) {
+            nextSibling = getNextNode(nextSibling);
+        }
+
+        if (!isLeadingAsterisk(nextSibling) && !isLastLine(nextSibling)) {
+            log(nextSibling.getLineNumber(), MSG_MISSING_ASTERISK);
+        }
+    }
+
+    /**
+     * Gets next node in the ast (sibling or parent sibling for the last node).
+     *
+     * @param detailNode the node to process
+     * @return next node.
+     */
+    private static DetailNode getNextNode(DetailNode detailNode) {
+        DetailNode node = JavadocUtil.getFirstChild(detailNode);
+        if (node == null) {
+            node = JavadocUtil.getNextSibling(detailNode);
+            if (node == null) {
+                DetailNode parent = detailNode;
+                do {
+                    parent = parent.getParent();
+                    node = JavadocUtil.getNextSibling(parent);
```

Comment:
This whole method looks weird. I would have thought it would look something similar to how we traverse through the AST.

My main question is the do loop. This shouldn't be a nested call to `getNextNode`? We truthfully only look 1 child deep?

Reply:
This is workaround for #9005 where `NEWLINE` may be too deep in the AST, for example:
```java
  /**
   *
     @deprecated block tag
   */
```
AST is
```
|   |--BLOCK_COMMENT_BEGIN -> /* [1:2]
|   |   |--COMMENT_CONTENT -> *\n   *\n     @deprecated block tag\n    [1:4]
|   |   |   `--JAVADOC -> JAVADOC [1:5]
|   |   |       |--NEWLINE -> \n [1:5]
|   |   |       |--LEADING_ASTERISK ->    * [2:0]
|   |   |       |--NEWLINE -> \n [2:4]
|   |   |       |--TEXT ->       [3:0]
|   |   |       |--JAVADOC_TAG -> JAVADOC_TAG [3:5]
|   |   |       |   |--DEPRECATED_LITERAL -> @deprecated [3:5]
|   |   |       |   |--WS ->   [3:16]
|   |   |       |   `--DESCRIPTION -> DESCRIPTION [3:17]
|   |   |       |       |--TEXT -> block tag [3:17]
|   |   |       |       |--NEWLINE -> \n [3:26]                         // NEWLINE HERE
|   |   |       |       `--TEXT ->     [4:0]                                  // Blank text, ignored in isLastLine
|   |   |       `--EOF -> <EOF> [4:3]                                   // Next node, that is next sibling of grandparent
|   |   `--BLOCK_COMMENT_END -> */ [4:3]
```
If #9005 can be solved, your comment will be correct: no more than one node deep